bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -277,122 +281,13 @@ private void checkEnvironment(RegistrationManager rm) throws BookieException, IO
-                // that dir is empty. Else, we reject the request
-                Set<String> existingLedgerDirs = Sets.newHashSet();
-                for (Cookie journalCookie : journalCookies) {
-                    Collections.addAll(existingLedgerDirs, journalCookie.getLedgerDirPathsFromCookie());
-                }
-                List<File> dirsMissingData = new ArrayList<File>();
-                List<File> nonEmptyDirs = new ArrayList<File>();
-                for (File dir : missedCookieDirs) {
-                    if (existingLedgerDirs.contains(dir.getParent())) {
-                        // if one of the existing ledger dirs doesn't have cookie,
-                        // let us not proceed further
-                        dirsMissingData.add(dir);
-                        continue;
-                    }
-                    String[] content = dir.list();
-                    if (content != null && content.length != 0) {
-                        nonEmptyDirs.add(dir);
-                    }
-                }
-                if (dirsMissingData.size() > 0 || nonEmptyDirs.size() > 0) {
-                    LOG.error("Either not all local directories have cookies or directories being added "
-                            + " newly are not empty. "
-                            + "Directories missing cookie file are: " + dirsMissingData
-                            + " New directories that are not empty are: " + nonEmptyDirs);
-                    throw new BookieException.InvalidCookieException();
-                }
-            }
-
-            if (missedCookieDirs.size() > 0) {
-                LOG.info("Stamping new cookies on all dirs {}", missedCookieDirs);
-                for (File journalDirectory : journalDirectories) {
-                    masterCookie.writeToDirectory(journalDirectory);
-                }
-                for (File dir : allLedgerDirs) {
-                    masterCookie.writeToDirectory(dir);
-                }
-                masterCookie.writeToRegistrationManager(rm, conf, rmCookie != null ? rmCookie.getVersion() : Version.NEW);
-            }
-
-            List<File> ledgerDirs = ledgerDirsManager.getAllLedgerDirs();
-            checkIfDirsOnSameDiskPartition(ledgerDirs);
-            List<File> indexDirs = indexDirsManager.getAllLedgerDirs();
-            checkIfDirsOnSameDiskPartition(indexDirs);
-            checkIfDirsOnSameDiskPartition(journalDirectories);
+        checkEnvironmentWithStorageExpansion(conf, rm, journalDirectories, allLedgerDirs);
 
-        } catch (IOException ioe) {
-            LOG.error("Error accessing cookie on disks", ioe);
-            throw new BookieException.InvalidCookieException(ioe);
-        }
+        List<File> ledgerDirs = ledgerDirsManager.getAllLedgerDirs();
+        checkIfDirsOnSameDiskPartition(ledgerDirs);
+        List<File> indexDirs = indexDirsManager.getAllLedgerDirs();
+        checkIfDirsOnSameDiskPartition(indexDirs);
```

Comment:
Why not simply call checkIfDirsOnSameDiskPartition(allLedgerDirs)?

Reply:
we can do that. I didn't touch this to keep the original logic unchanged.

Reply:
I changed to check (allLedgerDirs)